ProcessMaker/Repositories/TokenRepository.php

```diff
@@ -89,7 +89,26 @@ public function persistActivityActivated(ActivityInterface $activity, TokenInter
         $this->addUserToData($token->getInstance(), $user);
         $this->addRequestToData($token->getInstance());
         $token->user_id = $user ? $user->getKey() : null;
-        $token->is_self_service = $token->getAssignmentRule() === 'self_service' ? 1 : 0;
```

Comment:
I suggest that all this assignment logic should be in the `getNextUser()` method and here something like
```
    $token->is_self_service = $token->getAssignmentRule() === 'self_service' && $user === null ? 1 : 0;
```

Reply:
@caleeli You are right, the presence of $user combined with getAssignmentRule() is sufficient to determine the is_self_service flag. Updated.